x-pack/plugins/ingest_manager/server/services/epm/packages/install.ts

```diff
@@ -156,25 +159,44 @@ export async function saveInstallationReferences(options: {
   pkgName: string;
   pkgVersion: string;
   internal: boolean;
-  toSave: AssetReference[];
+  toSaveRefs: AssetReference[];
+  toSaveESIndexPatterns: Record<string, string>;
 }) {
-  const { savedObjectsClient, pkgName, pkgVersion, internal, toSave } = options;
+  const {
+    savedObjectsClient,
+    pkgName,
+    pkgVersion,
+    internal,
+    toSaveRefs,
+    toSaveESIndexPatterns,
+  } = options;
   const installation = await getInstallation({ savedObjectsClient, pkgName });
-  const savedRefs = installation?.installed || [];
+  const savedRefs = installation?.installed.references || [];
+  const toInstallESIndexPatterns = Object.assign(
+    installation?.installed.es_index_patterns || {},
```

Comment:
does the es_index_patterns need to be in the installed attribute?  since it isn't really installed and never uninstalled, but just becomes associated with an index pattern that already exists, i'm not sure it makes sense.  Then we could keep this function returning AssetReference[] and not have to make changes in the remove.ts file.

Reply:
Yeah good point, I'll move `es_index_patterns` to be at the same level as `installed`

```typescript
export interface Installation extends SavedObjectAttributes {
  installed: AssetReference[];
  es_index_patterns: Record<string, string>;
  name: string;
  version: string;
}
```